### DISCUSSION TITLE
mein openDTU versteht mich nicht. die zweite

### DISCUSSION THREAD

**Discussion by 3aau** [tbnobody/OpenDTU#1624] [Q&A]
moin,
ich komme hier immer noch nicht so richtig weiter.
ich habe einen kleinen datenlogger (tennsy4.1) der die mqtt daten empfängt, dieser soll nun auch dem opendtu per mqtt infos senden.
ich möchte das limit des wr steuern.
per HTTP sende ich das paket "0x30,0x37,0x00,0x35,solar/116183925733/cmd/limit_nonpersistent_relative47" an die openDTU.
im nächsten datensatz steht: solar/dtu/statustrue
protokol nach MQTT Version 3.1.1
    PUBLISH (1byte)
        restlänge (1byte)
          UTF-8 stringlänge (2byte)
           string...............................
ich im monitor der openDTU sehe ich:
14:12:08.823 > Nothing received, resend whole request
14:12:08.823 > TX AlarmData Channel: 75 --> 15 83 92 57 49 80 13 52 88 80 11 00 65 99 51 A0 00 00 00 00 00 00 00 00 81 4F 01 
14:12:09.297 > Connected to MQTT.
14:12:09.484 > Received MQTT message on topic: solar/116183925733/cmd/limit_nonpersistent_relative47
14:12:09.619 > RX Period End
14:12:09.619 > All missing
14:12:09.619 > Nothing received, resend whole request
14:12:09.619 > TX AlarmData Channel: 3 --> 15 83 92 57 49 80 13 52 88 80 11 00 65 99 51 A0 00 00 00 00 00 00 00 00 81 4F 01 
14:12:09.699 > Interrupt received
14:12:09.756 > RX Channel: 75 --> 95 83 92 57 49 83 92 57 49 01 00 01 B0 01 00 01 0B F7 0B F7 00 00 00 00 B0 02 97 | -80 dBm
14:12:09.819 > Interrupt received
14:12:09.902 > RX Channel: 40 --> 95 83 92 57 49 83 92 57 49 02 00 4E 0F C2 0F C2 00 00 00 05 B0 02 00 4F 0F C7 E9 | -80 dBm
14:12:10.004 > Interrupt received
14:12:10.052 > RX Channel: 40 --> 95 83 92 57 49 83 92 57 49 03 0F C7 FF FF FF FB B0 02 00 50 0F CC 0F CC 00 00 B8 | -80 dBm
14:12:10.209 > Interrupt received
14:12:10.313 > RX Channel: 40 --> 95 83 92 57 49 83 92 57 49 04 00 05 B0 02 00 51 0F DB 0F DB FF FF FF FB B0 02 C1 | -80 dBm
14:12:10.364 > Interrupt received
14:12:10.413 > RX Channel: 23 --> 95 83 92 57 49 83 92 57 49 05 00 52 0F E0 0F E0 00 00 00 05 B0 02 00 53 10 03 35 | -80 dBm
14:12:10.522 > Interrupt received
14:12:10.619 > RX Channel: 61 --> 95 83 92 57 49 83 92 57 49 08 00 56 10 56 10 56 00 00 00 05 B0 02 00 57 10 83 B8 | -80 dBm
14:12:10.723 > Interrupt received
14:12:10.773 > RX Channel: 61 --> 95 83 92 57 49 83 92 57 49 0A 00 05 B0 02 00 59 10 97 10 97 FF FF FF FB B0 02 C7 | -80 dBm
14:12:10.832 > RX Period End
14:12:10.832 > Last missing
14:12:10.832 > Request retransmit: 11
14:12:10.832 > TX RequestFrame Channel: 23 --> 15 83 92 57 49 80 13 52 88 8B D8 
14:12:10.880 > RX Period End
aber der wr bleibt bei seiner vorherigen einstellung des limit.

danke schon mal

**Comment by stefan123t** [comment]
@3aau hier ist the offizielle Dokumentation für die MQTT topics:

https://tbnobody.github.io/OpenDTU-docs/firmware/mqtt_topics/#inverter-limit-specific-topics

```
14:12:09.484 > Received MQTT message on topic: solar/116183925733/cmd/limit_nonpersistent_relative47
```

Das weitere Logfile deutet darauf hin dass er aktuell sich mit den Alert/EventLog des WR noch beschäftigt und daher noch kein Limit erfolgreich annimmt.

Eventuell liegt es aber auch am MQTT Broker / Client siehe hier

https://github.com/tbnobody/OpenDTU/issues/35#issuecomment-1274993866

**Reply by 3aau** [reply]
danke für die hilfe.
hatte wohl die länge des uft-8 strings falsch berechnet.
im monitor des opendtu kommt nun das hier. 
----------------------------
22:48:04.157 > TX AlarmData Channel: 3 --> 15 83 92 57 49 80 13 52 88 80 11 00 65 99 CA 90 00 00 00 00 00 00 00 00 18 6C 10 
22:48:04.361 > Received MQTT message on topic: solar/116183925733/cmd/limit_nonpersistent_absolute
22:48:04.361 > Limit Non-Persistent: 391 W
22:48:04.949 > RX Period End
---------------------------
schaut schon etwas besser aus. mal schauen wie es morgen mit laufendem wr geht.

ok. nun gehts. ich kann das limit per mqtt einstellen.
die länge des uft-8 string wird ohne den limitwert berechnet. steht leider nicht so in der beschreibung des mqtt protokol.
nun steuert mein kleiner datenlogger den wr.

als nächstes werde ich mich mal an die zeitsteuerung machen. 
grundidee ist 
1.) wr geht bei einschalten auf einen minimalwert also limit_nonpersistent_absolute z.b. 10 watt.
     damit bei irgendwelchen problemen die batterie nicht sofort leer gesaugt wird.
2.) ist NICHT tag (von bis) wird wr auf limit_nonpersistent_relative Pnacht gesetzt (z.b. 150 watt)
3.) ist tag (von bis) wird wr auf limit_nonpersistent_relative Ptag gesetzt (z.b. 600 watt)
     ist tag und panel liefert weniger als Pnacht (wolke) wird wr auf limit_nonpersistent_relative Pnacht gesetzt (z.b. 150 watt)
4.) ist der ertrag vom vortag kleiner  als akku kapazität wird Pnacht entsprechend reduziert
5.) ???????????????? noch wer ideen ?????????????
